### PR TITLE
Remove nrfxxx nwtsim

### PR DESCRIPTION
**Pull request by galak**
CONFIG_BOARD_NRFXX_NWTSIM doesn't exist and thus any code ifdef
protected cant be compiled or tested.  Remove this dead code.

Additionally, this code scatters a ton of CONFIG_BOARD_* defines in drivers and core code and we shouldn't be doing that.

### CODE REVIEW

**Review by carlescufi — CHANGES_REQUESTED** [review]
Blocking merge until @aescolar-ot has had a chance to propose an alternative.

**Comment by galak** [comment]
> This requires some discussion. In particular whether it would be possible to have this encapsulated in a more centralized way where it wouldn't be scattered all over the code, so that compatibility with the non-existent board can be kept upstream but at the same time it doesn't tarnish the code in the way it's today.
> 
> @aescolar-ot Do you have any proposals in this direction?

Agreed, as a general rule we shouldn't accept changes that have CONFIG_BOARD_* in subsytems and drivers w/o some really good reason, and especially not over 50 #ifdef/ifndef cases.

**Comment by carlescufi** [comment]
>Agreed, as a general rule we shouldn't accept changes that have CONFIG_BOARD_* in subsytems and drivers w/o some really good reason, and especially not over 50 #ifdef/ifndef cases.

Maybe we could introduce a new Kconfig option that targets special simulated environments so that instead of relying on a particular board, we add this knowledge directly into the kernel.

**Comment by galak** [comment]
> Maybe we could introduce a new Kconfig option that targets special simulated environments so that instead of relying on a particular board, we add this knowledge directly into the kernel.

Even for simulators we should try and avoid this kinda thing.  I know we've done it a little in a few places for qemu or other simulators, but I think it needs to be very isolated.

I'm not sure if anyone can share what's going on with the NRF_*_regw_sideeffects* calls.  What are they trying to accomplish?

**Review by aescolar — CHANGES_REQUESTED** [review]
Thanks @carlescufi . Indeed this requires some discussion.
@galak please allow us a couple of days to discuss this and present an appropriate way forward.

**Comment by galak** [comment]
> @galak please allow us a couple of days to discuss this and present an appropriate way forward.

No worries, I will.  Was opening this PR so people can see the code impact and generate the discussion on the topic.

**Comment by SebastianBoe** [comment]
Some snippets from the last time this was discussed:

> Running the same SW on HW and in a simulator is a very old problem.
>
> There exist some more sofisticated solutions than what we have done here that we could be inspired by.
>
> They pretty much all require that we interface with some function-based interface (HAL layer) instead of this unmanage-able struct-based interface (CMSIS).

> If you want to avoid macros you could always have a single header-file with
> the function prototypes and then 2 different source implementations, one for sim,
> one for HW. This is a zero-cost abstraction if you build it correctly.

**Comment by galak** [comment]
What's the latest on the cleanup here?  Its been a few weeks now.

**Comment by aescolar-ot** [comment]
@galak A more thorough cleanup is ready in a provisional repo using the NRFx HAL.
But that required a few fixes in the NRFx HAL, some of which are in the process of being merged in right now.
As soon as those are in, another PR will follow here.

You can safely close this PR.

**Comment by galak** [comment]
Want to leave this open til the other PRs mentioned get merged.

**Comment by galak** [comment]
What are the PR # that fix this?

**Comment by carlescufi** [comment]
@galak no PRs yet, we are waiting for the required drivers to get merged to nrfx first.

**Comment by galak** [comment]
@carlescufi what timeframe are we talking about?  If this is going to take some time, I'd rather we remove this dead code for now, and let a proper fix come in when you guys are ready.

**Comment by carlescufi** [comment]
@galak target is next week.

**Comment by aescolar-ot** [comment]
@galak this code is not dead, we are using it. It is just of no direct use for you.
Anyhow, the commits for fixing this are ready and based on the original code. 
If this PR were to be merged, it would create unnecessary extra work for the rest of us. Given that this code is not creating any real inconvenience for you, I would hope you could wait.

**Comment by galak** [comment]
> @galak this code is not dead, we are using it. It is just of no direct use for you.
> Anyhow, the commits for fixing this are ready and based on the original code.
> If this PR were to be merged, it would create unnecessary extra work for the rest of us. Given that this code is not creating any real inconvenience for you, I would hope you could wait.

The code is dead since it can't be built or tested by any random person.

I'm fine waiting as long as this get resolved before the merge window for 1.12 closes.

**Review by pskhansen — CHANGES_REQUESTED** [review]
nope

**Review by cvinayak — CHANGES_REQUESTED** [review]
There is work-in-progress by @aescolar-ot to address the concern of dead code by using nRF HAL from ext folder, simulator board (any other defined boards) can choose to compile-in different HAL without the need for conditional compilations.

**Comment by carlescufi** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/8418 fixes this instead

**Pull request closed without merging**
